netbox_librenms_plugin/templates/netbox_librenms_plugin/htmx/device_validation_details.html

```
@@ -198,7 +198,82 @@ <h6 class="card-header d-flex justify-content-between align-items-center">
-                    <span class="text-danger"><i class="mdi mdi-close-circle"></i> No matching type</span>
+                    {% if libre_device.hardware and libre_device.hardware != "-" %}
+                      <div>
+                        <span class="text-danger d-block"><i class="mdi mdi-close-circle"></i> No matching type</span>
+                        <form hx-post="{% url 'plugins:netbox_librenms_plugin:add_device_type_mapping' device_id=libre_device.device_id %}"
+                              hx-swap="none"
+                              hx-include="[name=role_{{ libre_device.device_id }}], [name=rack_{{ libre_device.device_id }}], [name=cluster_{{ libre_device.device_id }}], #use-sysname-toggle, #strip-domain-toggle"
+                              class="mt-1"
+                              id="dt-mapping-form-{{ libre_device.device_id }}">
+                          {% csrf_token %}
+                          <div class="position-relative">
+                            <input type="text"
```

HTMLHint (1.9.2)

[warning] No matching [ label ] tag found.

(input-requires-label)